signimage/signimage.go

```diff
@@ -0,0 +1,357 @@
+	}
+	f.Close()
+
+	return nil
+
+}
+
+func getAuthFromFile(configfile string, repo string) (authn.Authenticator, error) {
+
+	if configfile == "" {
+		fmt.Printf("\n== no pull secret defined, default to Anonymous\n")
+		return authn.Anonymous, nil
+	}
+
+	f, err := os.Open(configfile)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+	cf, err := config.LoadFromReader(f)
+	if err != nil {
+		return nil, err
+	}
+
+	var cfg dockertypes.AuthConfig
+
+	cfg, err = cf.GetAuthConfig(repo)
+	if err != nil {
+		return nil, err
+	}
+
+	return authn.FromConfig(authn.AuthConfig{
+		Username:      cfg.Username,
+		Password:      cfg.Password,
+		Auth:          cfg.Auth,
+		IdentityToken: cfg.IdentityToken,
+		RegistryToken: cfg.RegistryToken,
+	}), nil
+
+}
+
+func main() {
+	// get the env vars we are using for setup, or set some sensible defaults
+
+	var unsignedimagename string
+	var signedimagename string
+	var fileslist string
+	var privkeyfile string
+	var pubkeyfile string
+	var pullsecret string
+	var pushsecret string
+	var nopush bool
+	flag.StringVar(&unsignedimagename, "unsignedimage", "", "name of the image to sign")
+	flag.StringVar(&signedimagename, "signedimage", "", "name of the signed image to produce")
+	flag.StringVar(&fileslist, "filestosign", "", "colon seperated list of kmods to sign")
+	flag.StringVar(&privkeyfile, "key", "", "path to file containing private key for signing")
+	flag.StringVar(&pubkeyfile, "cert", "", "path to file containing public key for signing")
+	flag.StringVar(&pullsecret, "pullsecret", "", "path to file containing credentials for pulling images")
+	flag.StringVar(&pullsecret, "pushsecret", "", "path to file containing credentials for pushing images")
+	flag.BoolVar(&nopush, "no-push", false, "do not push the resulting image")
+
+	flag.Parse()
+
+	checkarg(&unsignedimagename, "UNSIGNEDIMAGE", "")
+	checkarg(&signedimagename, "SIGNEDIMAGE", unsignedimagename+"signed")
+	checkarg(&fileslist, "FILESTOSIGN", "")
+	checkarg(&privkeyfile, "KEYSECRET", "")
+	checkarg(&pubkeyfile, "CERTSECRET", "")
+	checkarg(&pullsecret, "PULLSECRET", "")
+	checkarg(&pushsecret, "PUSHSECRET", pullsecret)
+	// if we've made it this far the arguements are sane
+
+	// get a temp dir to copy kmods into for signing
+	extractiondir, err := os.MkdirTemp("/tmp/", "kmod_signer")
+	if err != nil {
+		fmt.Errorf("could not create temp dir: %v\n", err)
+		panic(err)
+	}
+
+	// sets up a tar archive we will use for a new layer
+	var b bytes.Buffer
+	tarwriter := tar.NewWriter(&b)
+
+	//make a map of the files to sign so we can track what we want to sign
+	kmodstosign := make(map[string]string)
+	for _, x := range strings.Split(fileslist, ":") {
+		kmodstosign[x] = "not found"
+	}
+
+	// set up image download otions
+	opts := make([]name.Option, 0)
+	ref, err := name.ParseReference(unsignedimagename, opts...)
+	if err != nil {
+		fmt.Errorf("could not parse the container image name: %v\n", err)
+		panic(err)
```

Comment:
What does that line do? it is just returning a new error but no one is using it. Am I missing something?
```suggestion
		panic(fmt.Errorf("could not parse the container image name: %w", err))
```